Revert "Revert "Require jackson2-2.12.6.1""

baldersheim opened the pull request:
@hmusum PR
Trying this one again

baldersheim committed:
Revert "Revert "Require jackson2-2.12.6.1""

baldersheim committed:
Exclude jakarta.activation-api and  jakarta.xml.bind-api from jackson-jaxrs-json-provider

baldersheim commented:
@bjorncs or @gjoranv Can you verify if this is the correct way of excluding jakarta stuff ?

gjoranv commented:
I'm looking into this now. Please don't merge.

gjoranv commented on container-dependency-versions/pom.xml:
```diff
@@ -71,6 +71,16 @@
                 <groupId>com.fasterxml.jackson.jaxrs</groupId>
                 <artifactId>jackson-jaxrs-json-provider</artifactId>
                 <version>${jackson2.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>jakarta.activation</groupId>
+                        <artifactId>jakarta.activation-api</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>jakarta.xml.bind</groupId>
+                        <artifactId>jakarta.xml.bind-api</artifactId>
+                    </exclusion>
+                </exclusions>
```
Jakarta should already be excluded in the necessary pom files, so this should be unnecessary. I have verified that neither `container-dev` nor `container` lead jakarta if these exclusions are removed. Also, if a user adds an explicit dependency on jackson-jaxrs-json-provider, these exclusions will take effect for them. We want to affect user deps as little as possible.

BTW, the jackson-jaxrs-json-provider will probably be completely removed from Vespa 8.

baldersheim replied:
But I would assume that if you added a dependency that is provided, you would not need to add any exclusions for it yourself. Then it should be done by the provider.
I have forund 3 places we do that ourselves.

gjoranv replied:
No, we don't want to interfere with users' dependencies any more than we have to. We are actively trying to reduce any overlap between our maven deps and the users'. This will be gone on Vespa 8 anyway. I will add commits for what I consider the best workaround.

gjoranv replied:
Users depend on `container`, where jakarta is excluded. If they for some reason want to add their own dependencies, we should not interfere.

gjoranv committed:
Do not add exclusions in container-dep-versions

gjoranv committed:
Add explicit jackson-module-jaxb-annotations to exclude jakarta.

- Necessary since jackson was upgraded from 2.12 to 2.13.
- This will be removed for Vespa 8
